napi.h

```diff
@@ -2061,6 +2062,65 @@ namespace Napi {
     std::unique_ptr<napi_threadsafe_function, Deleter> _tsfn;
     Deleter _d;
   };
+
+  template<class T>
+  class AsyncProgressWorker : public AsyncWorker {
+    public:
+     virtual ~AsyncProgressWorker();
+
+     class ExecutionProgress {
```

Comment:
I assume this separate class is used so that the Execute method only has access to the methods which are safe to call right?

Reply:
Yes, the ExecutionProgress was created on the stack, and all its methods were safe to call on the worker thread.